Brauser.Common/Recipe/AdditionalFermentationIngredient.cs

```diff
@@ -2,6 +2,7 @@ namespace Brauser.Common.Recipe;
 
 public class AdditionalFermentationIngredient
 {
+    public Guid Id { get; set; }
```

Comment:
What's our plan with the Guids in the single ingredients? Do we plan to keep another list of all possible things?

Reply:
I assume, that we will have ingredient lists with matching GUIDs.
In that case, we should open an issue for that, since we do need extra lookup while converting or creating recipes.

Reply:
The plan for the GUIDs is just to make editing the ingredients easier. When editing the things in the frontend, I need a way to uniquely identify the ingredient that should be updated. I don't think we need an extra list of ingredients with the same GUIDs. Because the `Order`, `Quantity`, and `Unit` might be different every time. And just for the name an extra list is a little too much.

Reply:
This means the converter could just call `Guid.NewGuid()` for every ingredient it creates for a recipe.

Reply:
Since we will use a document based db (most likely mongo) I still don't see the benefits. It's pretty easy to save the whole recipe in there and also for editing it should be pretty easy. Extra fields such as Guids would be nice for referencing, but I don't think we do need this for other cases

Reply:
Ah, ok. Looks like I wasn't really clear on what I meant.
Yes, we don't **need** the (GU)ID for the database at all
When I say "update", I'm referring to the update that is done in the object inside the (.NET-)backend. Not in the database.

Example: I have an existing recipe with two ingredients.
```json
{
  "Id": 123,
  // ...
  // a lot of recipe information
  // ...
  "AdditionalFermentationIngredient": [
    {
      "Order": 1,
      "Name": "Toller Zusatz",
      "Quantity": 2,
      "Unit": "kg"
    },
    {
      "Order": 2,
      "Name": "Noch ein Zusatz",
      "Quantity": 100,
      "Unit": "g"
    }
  ]
}
```

Now I open the recipe editor and click on "edit" for the ingredient with the name "Toller Zusatz". I change the `Order` to "2" and the `Name` to "Noch ein Zusatz" and click the "Save"-Button. The backend now receives an object that looks like this:
```json
    {
      "Order": 2,
      "Name": "Noch ein Zusatz",
      "Quantity": 2,
      "Unit": "kg"
    },
```

How does the backend decide which of the two ingredients should be updated? That's the main reason I suggested this change. I'm sure there might be another smart way to achieve this, but this is the way I usually do it. 😂🤷

Reply:
Maybe I could just send the entire recipe every time something changes?

Reply:
imho we would have to update multiple ingredients when the order changes. Therefore, we should at least update the complete ingredient list, right?

But maybe we should discuss keeping an additional database for ingredients. Would make it possible to check for recipes based on the own inventory?

Reply:
> But maybe we should discuss keeping an additional database for ingredients. Would make it possible to check for recipes based on the own inventory?

Good point: We should keep that in mind.